hazelcast/src/main/java/com/hazelcast/transaction/impl/xa/XATransactionContextImpl.java

```diff
@@ -118,7 +124,7 @@ public TransactionalObject getTransactionalObject(String serviceName, String nam
 
         final Object service = nodeEngine.getService(serviceName);
         if (service instanceof TransactionalService) {
-            nodeEngine.getProxyService().initializeDistributedObject(serviceName, name);
+            nodeEngine.getProxyService().initializeDistributedObject(serviceName, name, source);
```

Comment:
Same as https://github.com/hazelcast/hazelcast/pull/16347/files#r361291907

Reply:
It is filled with null for server side calls again, hence the same reasoning as above.